### PR TITLE
Create subinterpreter example

### PR DESCRIPTION
**Pull request by mejrs**
I wrote up most of this when I was thinking about how we can support these in pyo3. I figured I might as well share it.

### CODE REVIEW

File: examples/sequential/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[build-system]
+requires = ["maturin>=1,<2"]
+build-backend = "maturin"
+
+[project]
+name = "string sum"
```
Comment (adamreichold) [inline examples/sequential/pyproject.toml:6]: Copy pasta.

File: examples/sequential/requirements-dev.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+pytest>=3.5.0
+pip>=21.3
+maturin>=0.12,<0.13
```
Comment (adamreichold) [inline examples/sequential/requirements-dev.txt:3]: Seems redundant to the Nox session and inconsistent with pyproject.toml? Maybe drop the requirements file outright and rely solely on the Nox session?

Comment (davidhewitt) [reply examples/sequential/requirements-dev.txt:3]: Probably there's a broader tidy up we should consider for all of these examples, I suspect this was copypasta too.

**Review by davidhewitt — COMMENTED** [review]
Thanks, this is cool to see! When it's laid out like this, it doesn't seem so hard right? Just need to figure out how to get our nice safe high-level wrappers to be sound under these conditions 😂

File: examples/sequential/requirements-dev.txt (hunk 2 above)
Comment (davidhewitt) [reply examples/sequential/requirements-dev.txt:3]: Resolved separately in #3536, I'll rebase this.

**Review by davidhewitt — APPROVED** [review]
I rebased this and fixed CI, looks good to me, will merge!

**Comment by davidhewitt** [comment]
Fixed up for MSRV.

**Comment by mejrs** [comment]
Thanks :)

**Pull request merged**